packages/flutter_tools/lib/src/android/deferred_components_setup_validator.dart

```diff
@@ -0,0 +1,766 @@
+///
+/// To configure a setup verification run, supply a list of DeferredComponentsSetupValidatorCheck
+/// values to the contstructor. To run the the setup verification, invoke [runSetup] which
+/// will run each check in the order provided. The results of the checks will be displayed.
+///
+/// Failing a check will result in the tool to exit, ending the current command unless false is
+/// passed to the [exitOnFail] parameter.
+class DeferredComponentsSetupValidator {
+  DeferredComponentsSetupValidator(this.env, {
+    this.exitOnFail = true,
+    String title,
+    Directory templatesDir,
+  }) : _outputDir = env.projectDir
+      .childDirectory('build')
+      .childDirectory(kDeferredComponentsTempDirectory),
+      _inputs = <File>[],
+      _outputs = <File>[],
+      _title = title ?? 'Deferred components setup verification',
+      _templatesDir = templatesDir,
+      _generatedFiles = <String>[],
+      _modifiedFiles = <String>[],
+      _diffLines = <String>[];
+
+  final Environment env;
+
+  /// When true, failed checks and tasks will result in [attemptToolExit] triggering
+  /// [throwToolExit].
+  final bool exitOnFail;
+
+  /// The name of the golden file that tracks the latest loading units generated.
+  @visibleForTesting
+  static const String kDeferredComponentsGoldenFileName = 'deferred_components_golden.yaml';
+  /// The directory in the build folder to generate missing/modified files into.
+  @visibleForTesting
+  static const String kDeferredComponentsTempDirectory = 'android_deferred_components_setup_files';
+
+  final String _title;
+  final Directory _templatesDir;
+  final Directory _outputDir;
+  // Files that were newly generated by this validator.
+  final List<String> _generatedFiles;
+  // Existing files that were modified by this validator.
+  final List<String> _modifiedFiles;
+  // Output of the diff task.
+  // TODO(garyq): implement the diff task.
+  final List<String> _diffLines;
+  // Tracks the new and missing loading units.
+  Map<String, dynamic> _goldenComparisonResults;
+  // Input and output Files used to generate a Depfile.
+  final List<File> _inputs;
+  final List<File> _outputs;
+
+  /// Returns a [Depfile] that includes all input dependencies and non-temporary output
+  /// files.
+  Depfile get depfile => Depfile(_inputs, _outputs);
+
+  /// Returns true if there were any recommended changes that should
+  /// be applied.
+  ///
+  /// Retuns false if no problems were detected.
+  ///
+  /// If no checks are run, then this will default to false and will remain so
+  /// until a failing check finishes running.
+  bool get changesNeeded => _generatedFiles.isNotEmpty || _modifiedFiles.isNotEmpty
+    || (_goldenComparisonResults != null && !(_goldenComparisonResults['match'] as bool));
+
+  /// Checks if all dynamic feature modules are included in `settings.gradle`.
+  ///
+  /// `android/settings.gradle` should `include` all of the dynamic feature modules.
```

Comment:
Can you link to an example/docs on what structure this is looking for in settings.gradle?